src/benchmark/tpch_benchmark.cpp

```diff
@@ -0,0 +1,355 @@
+        SQLPipeline{sql}.get_result_table();
+      }
+
+      QueryBenchmarkResult result;
+      result.num_iterations = state.num_iterations;
+      result.duration = std::chrono::high_resolution_clock::now() - state.begin;
+
+      _query_results_by_query_id.emplace(query_id, result);
+    }
+  }
+
+  // Create a report in roughly the same format as google benchmarks do when run with --benchmark_format=json
+  void _create_report(std::ostream& stream) const {
+    nlohmann::json benchmarks;
+
+    for (const auto query_id : _query_ids) {
+      const auto& query_result = _query_results_by_query_id.at(query_id);
+
+      const auto duration_ns = std::chrono::duration_cast<std::chrono::nanoseconds>(query_result.duration).count();
+      const auto duration_seconds = static_cast<float>(duration_ns) / 1'000'000'000;
+      const auto items_per_second = static_cast<float>(query_result.num_iterations) / duration_seconds;
+      const auto time_per_query = duration_ns / query_result.num_iterations;
+
+      nlohmann::json benchmark{
+          {"name", "TPC-H " + std::to_string(query_id + 1)},
+          {"iterations", query_result.num_iterations},
+          {"real_time", time_per_query},
+          {"cpu_time", time_per_query},
+          {"items_per_second", items_per_second},
+          {"time_unit", "ns"},
+      };
+
+      benchmarks.push_back(benchmark);
+    }
+
+    /**
+     * Generate YY-MM-DD hh:mm::ss
+     */
+    auto current_time = std::time(nullptr);
+    auto local_time = *std::localtime(&current_time);
+    std::stringstream timestamp_stream;
+    timestamp_stream << std::put_time(&local_time, "%Y-%m-%d %H:%M:%S");
+
+    nlohmann::json context{
+        {"date", timestamp_stream.str()},
+        {"scale_factor", _scale_factor},
+        {"chunk_size", _chunk_size},
+        {"build_type", IS_DEBUG ? "debug" : "release"},
+        {"benchmark_mode",
+         _benchmark_mode == BenchmarkMode::IndividualQueries ? "IndividualQueries" : "PermutedQuerySets"}};
+
+    nlohmann::json report{{"context", context}, {"benchmarks", benchmarks}};
+
+    stream << std::setw(2) << report << std::endl;
+  }
+};
+
+}  // namespace opossum
+
+int main(int argc, char* argv[]) {
+  auto num_iterations = size_t{0};
+  auto timeout_duration = size_t{0};
+  auto scale_factor = 1.0f;
+  auto chunk_size = opossum::ChunkOffset(opossum::INVALID_CHUNK_OFFSET);
+  auto benchmark_mode_str = std::string{"IndividualQueries"};
+
+  cxxopts::Options cli_options_description{"TPCH Benchmark", ""};
+
+  cli_options_description.add_options()("help", "print this help message")(
+      "v,verbose", "Print log messages", cxxopts::value<bool>(opossum::verbose_benchmark)->default_value("false"))(
+      "s,scale", "Database scale factor (1.0 ~ 1GB)", cxxopts::value<float>(scale_factor)->default_value("0.001"))(
+      "r,runs", "Maximum number of runs of a single query",
+      cxxopts::value<size_t>(num_iterations)->default_value("1000"))(
```

Comment:
This might be one of the cases where `clang-format off` / `nolint` can help improving readability